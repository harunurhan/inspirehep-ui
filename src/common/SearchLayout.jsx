import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'antd';

import AggregationFiltersContainer from './containers/AggregationFiltersContainer';
import PaginationContainer from './containers/PaginationContainer';
import SortByContainer from './containers/SortByContainer';
import ResultsContainer from './containers/ResultsContainer';

class SearchLayout extends Component {
  render() {
    return (
      <div>
        <Row>
          <Col span={6}>
            <AggregationFiltersContainer />
          </Col>
          <Col span={18}>
            <Row>
              <Col span={12}>
                <PaginationContainer />
              </Col>
              <Col span={12}>
                <SortByContainer />
              </Col>
            </Row>
            <Row>
              <ResultsContainer
                renderItem={this.props.renderResultItem}
              />
            </Row>
          </Col>
        </Row>
      </div>
    );
  }
}

SearchLayout.propTypes = {
  renderResultItem: PropTypes.func.isRequired,
};

export default SearchLayout;
