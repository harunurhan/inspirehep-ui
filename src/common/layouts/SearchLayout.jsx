import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'antd';

import AggregationFiltersContainer from './../containers/AggregationFiltersContainer';
import PaginationContainer from './../containers/PaginationContainer';
import SortByContainer from './../containers/SortByContainer';
import ResultsContainer from './../containers/ResultsContainer';

class SearchLayout extends Component {
  render() {
    return (
      <Row type="flex" justify="space-between">
        <Col span={6}>
          <AggregationFiltersContainer />
        </Col>
        <Col span={18}>
          <PaginationContainer />
          <SortByContainer />
          <ResultsContainer renderItem={this.props.renderResultItem} />
        </Col>
      </Row>
    );
  }
}

SearchLayout.propTypes = {
  renderResultItem: PropTypes.func.isRequired,
};

export default SearchLayout;
