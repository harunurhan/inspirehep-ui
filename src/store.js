import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'react-router-redux';
/* eslint-disable import/no-extraneous-dependencies */
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
/* eslint-disable import/no-extraneous-dependencies */

import reducers from './reducers';
import http from './common/http';
import queryParamsParserMiddleware from './middlewares/queryParamsParser';
import persistUserStateMiddleware, {
  reHydrateRootStateWithUser,
} from './middlewares/persistUserState';

export const thunkMiddleware = thunk.withExtraArgument(http);

export const history = createHistory();
const reduxRouterMiddleware = routerMiddleware(history);

const getMiddleware = () => {
  if (process.env.NODE_ENV === 'production') {
    return applyMiddleware(
      reduxRouterMiddleware,
      queryParamsParserMiddleware,
      persistUserStateMiddleware,
      thunkMiddleware
    );
  }
  return applyMiddleware(
    reduxRouterMiddleware,
    queryParamsParserMiddleware,
    persistUserStateMiddleware,
    thunkMiddleware,
    createLogger()
  );
};

export const store = createStore(
  reducers,
  reHydrateRootStateWithUser(),
  composeWithDevTools(getMiddleware())
);
