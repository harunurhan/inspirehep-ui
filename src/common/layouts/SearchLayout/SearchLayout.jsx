import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'antd';
import { connect } from 'react-redux';

import AggregationFiltersContainer from './../../containers/AggregationFiltersContainer/AggregationFiltersContainer';
import PaginationContainer from './../../containers/PaginationContainer';
import SortByContainer from './../../containers/SortByContainer';
import ResultsContainer from './../../containers/ResultsContainer';
import NumberOfResultsContainer from '../../containers/NumberOfResultsContainer';
import LoadingOrChildren from '../../components/LoadingOrChildren';
import './SearchLayout.scss';

class SearchLayout extends Component {
  render() {
    const {
      renderResultItem,
      loading,
      loadingAggregations,
      withoutAggregations,
      withoutSort,
    } = this.props;
    return (
      <Row className="__SearchLayout__" gutter={32} type="flex" justify="start">
        <Col lg={8} xl={6} xxl={5}>
          {!withoutAggregations && (
            <LoadingOrChildren loading={loadingAggregations}>
              <AggregationFiltersContainer />
            </LoadingOrChildren>
          )}
        </Col>
        <Col lg={16} xl={15} xxl={14}>
          <LoadingOrChildren loading={loading}>
            <Row type="flex" align="middle" justify="end">
              <Col span={12}>
                <NumberOfResultsContainer />
              </Col>
              <Col className="alignRight" span={12}>
                {!withoutSort && <SortByContainer />}
              </Col>
            </Row>
            <Row>
              <Col span={24}>
                <ResultsContainer renderItem={renderResultItem} />
                <PaginationContainer />
              </Col>
            </Row>
          </LoadingOrChildren>
        </Col>
      </Row>
    );
  }
}

SearchLayout.propTypes = {
  renderResultItem: PropTypes.func.isRequired,
  withoutAggregations: PropTypes.bool,
  withoutSort: PropTypes.bool,
  loading: PropTypes.bool.isRequired,
  loadingAggregations: PropTypes.bool.isRequired,
};

SearchLayout.defaultProps = {
  withoutAggregations: false,
  withoutSort: false,
};

const stateToProps = state => ({
  loading: state.search.get('loading'),
  loadingAggregations: state.search.get('loadingAggregations'),
});

export default connect(stateToProps)(SearchLayout);
