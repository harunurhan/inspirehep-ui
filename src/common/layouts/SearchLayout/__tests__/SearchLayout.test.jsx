import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import { getStoreWithState, getStore } from '../../../../fixtures/store';
import SearchLayout from '../SearchLayout';

describe('SearchLayout', () => {
  it('renders with initial state', () => {
    const store = getStore();
    const wrapper = shallow(
      <SearchLayout store={store} renderResultItem={jest.fn()} />
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with loading states true', () => {
    const store = getStoreWithState(
      fromJS({
        search: {
          loading: true,
          loadingAggregations: true,
        },
      })
    );
    const wrapper = shallow(
      <SearchLayout store={store} renderResultItem={jest.fn()} />
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders without aggregations and sort', () => {
    const store = getStore();
    const wrapper = shallow(
      <SearchLayout
        store={store}
        withoutAggregations
        withoutSort
        renderResultItem={jest.fn()}
      />
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
