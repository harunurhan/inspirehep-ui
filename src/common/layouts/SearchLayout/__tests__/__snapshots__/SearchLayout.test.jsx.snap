// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchLayout renders with initial state 1`] = `
<Row
  className="__SearchLayout__"
  gutter={32}
  justify="start"
  type="flex"
>
  <Col
    lg={8}
    xl={6}
    xxl={5}
  >
    <LoadingOrChildren
      loading={false}
    >
      <Connect(AggregationFiltersContainer) />
    </LoadingOrChildren>
  </Col>
  <Col
    lg={16}
    xl={15}
    xxl={14}
  >
    <LoadingOrChildren
      loading={false}
    >
      <Row
        align="middle"
        gutter={0}
        justify="end"
        type="flex"
      >
        <Col
          span={12}
        >
          <Connect(NumberOfResultsContainer) />
        </Col>
        <Col
          className="alignRight"
          span={12}
        >
          <Connect(SortByContainer) />
        </Col>
      </Row>
      <Row
        gutter={0}
      >
        <Col
          span={24}
        >
          <Connect(ResultsContainer)
            renderItem={[Function]}
          />
          <Connect(PaginationContainer) />
        </Col>
      </Row>
    </LoadingOrChildren>
  </Col>
</Row>
`;

exports[`SearchLayout renders with loading states true 1`] = `
<Row
  className="__SearchLayout__"
  gutter={32}
  justify="start"
  type="flex"
>
  <Col
    lg={8}
    xl={6}
    xxl={5}
  >
    <LoadingOrChildren
      loading={false}
    >
      <Connect(AggregationFiltersContainer) />
    </LoadingOrChildren>
  </Col>
  <Col
    lg={16}
    xl={15}
    xxl={14}
  >
    <LoadingOrChildren
      loading={false}
    >
      <Row
        align="middle"
        gutter={0}
        justify="end"
        type="flex"
      >
        <Col
          span={12}
        >
          <Connect(NumberOfResultsContainer) />
        </Col>
        <Col
          className="alignRight"
          span={12}
        >
          <Connect(SortByContainer) />
        </Col>
      </Row>
      <Row
        gutter={0}
      >
        <Col
          span={24}
        >
          <Connect(ResultsContainer)
            renderItem={[Function]}
          />
          <Connect(PaginationContainer) />
        </Col>
      </Row>
    </LoadingOrChildren>
  </Col>
</Row>
`;
